torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -1513,6 +1513,42 @@ class IrParser {
             nullptr);
       }
     }
+    {
+      if (isOptionEnabled(EnableOption::GraphOp)) {
+        auto ptr_op = getOperatorForLiteral(
+            "aten::index_select(Tensor self, int dim, Tensor index) -> Tensor");
+        REGISTER_PARSE_RULE(
+            ptr_op,
+            {
+              MemoryFormat format;
+              std::list<Val*> list_val;
+              std::tie(format, list_val) = getPWFormatValues(
+                  c10::nullopt,
+                  value_map[node->inputs()[0]->unique()],
+                  value_map[node->inputs()[2]->unique()]);
+              auto input = list_val.front();
+              list_val.pop_front();
+              auto dim_value = constant_as<int>(node->input(1));
+              TORCH_INTERNAL_ASSERT(
+                  dim_value.has_value(), "dim in index_select is not valid");
+              auto index = list_val.front();
+              list_val.pop_front();
+              Val* out = nullptr;
```

Comment:
nitpick: Can we just put an `assert(false)` here with a `not implemented`? I suspect this will give us a segfault that's going to be confusing if index_select is somehow accidentally enabled.